feat(website): Add definitions to glossary (#318)
iron-fish/website: docs/onboarding/glossary.md

---
id: glossary
title: Glossary
sidebar_label: Glossary
description: Glossary | Iron Fish Onboarding
hide_table_of_contents: false
---

## Asset

A user-created resource which can be transferred privately, minted, and burned. Contains metadata, a name, the owner public address, and nonce. 

### Asset Generator

The [`group_hash`](#group-hash) of the [Hashed Asset Info](#hashed-asset-info). Calculated in public so the hash can be verified at consensus.

### Asset Identifier

The 32-byte representation of the [Asset Generator](#asset-generator) point.

### Hashed Asset Info

A Pedersen Hash of the Asset Info (owner, name, metadata, nonce) . This hash is the pre-image for the [`group_hash`](#group-hash) to generate the [Asset Generator](#asset-generator) point.

## Block

A piece of data in the Iron Fish chain which holds a record of transactions, graffiti, difficulty, previous [Block](#block), sequence, and timestamp.

### Graffiti

32 byte field that may be assigned at will by the miner who mined the block.

### Difficulty

Measure of how difficult it is to produce an Iron Fish [Block](#block).

### Sequence

> The sequential index number of a block in the Iron Fish chain, or the number of blocks preceding a given [Block](#block).

Reviewer: > the number of blocks preceding a given [Block](#block).

I think this might be slightly off -- I'd think of it like the number of blocks up to and including the current block (since the genesis sequence is 1)

## Circuit

A program used to create the [Proof](#proof) for a [Transaction Description](#description).

## Commitment

Product of a value and random number as a way of committing to a value without revealing the value itself. Generally a point on the elliptic curve.

## Groth16

The zero-knowledge proving system used in Iron Fish and Sapling. Named after the [paper](https://eprint.iacr.org/2016/260) by Jens Groth.

## Group Hash

A function which is used to obtain generator points on the Jubjub curve.

## Keys

A piece of data which can be used to encrypt and decrypt data.

### Spending Key

> Also known as Secret Key

32-byte random seed used to construct other parts of an Iron Fish wallet. The expanded form of this key is required before a note can be spent.

### Spend Authorization Key

> Also known as `ask`

Derived from spending key using a seeded pseudorandom hash function. Used to construct [Authorization Key](#authorization-key).

### Authorization Key

> Also known as `ak`

> Public key for the [Spend Authorization Key](#spend-authorization-key) by multiplying the [Spend Authorization Key](#spend-authorization-key) with a fixed generator base point for it on the Jubjub curve.

Reviewer: ```suggestion
Public key constructed by multiplying the [Spend Authorization Key](#spend-authorization-key) with a fixed generator base point on the Jubjub curve.
```

The repeated "Spend Authorization Key" felt like it might be a typo, maybe it could be worded like this?

### Proof Authorization Key

> Also known as `nsk`

Derived from spending key using a seeded pseudorandom hash function. Used to construct the Nullifier Deriving Key.

> ### Outgoing View Key

Reviewer: Is it worth adding the new `viewKey` composite key that's in the new account format in here?

### Fee

A reward in $IRON which is paid out to miners for confirming [Transactions](#transaction) to a [Block].

## Miners

Nodes that propose a new block to be added to the blockchain to other nodes.

## Mining

Core mechanism that defines by which rules new [Blocks](#block) are created and by which rules a peer can verify and accept an incoming [Blocks](#block).

## Note

The basic unit to represent value of any asset type on the Iron Fish chain. Always stored in its encrypted format publicly. Contains the owner address, original sender address, value, memo, and [Asset Generator](#asset-generator).

## Nullifier

A 32 byte hash that when revealed marks a note as spent so that note cannot be spent again.

## Proof

A method to prove that is statement is true to a verifier. Iron Fish proofs are 192 byte [Groth16](#groth16) that can be verified with the public parameters and the associated circuit’s prepared verifying key.

### Proof Inputs

The raw public and private parameters needed to create a [Proof](#proof).

## Wallet

An Iron Fish wallet manages [Accounts](#account), listens to incoming transactions from the chain, and broadcasts transactions to the chain.

### Account

> An Iron Fish account holds private [keys](#keys) which enables users to view transaction details and spend transfer assets to other addresses.

Reviewer: ```suggestion
An Iron Fish account holds private [keys](#keys) which enable users to view transaction details and transfer assets to other addresses.
```
